master/internal/api_tensorboard.go

```diff
@@ -80,20 +105,8 @@ func (a *apiServer) GetTensorboards(
 		return nil, err
 	}
 
-	a.filter(&resp.Tensorboards, func(i int) bool {
-		if err != nil {
-			return false
-		}
-		ok, serverError := command.AuthZProvider.Get().CanGetNSC(
-			ctx, *curUser, model.UserID(resp.Tensorboards[i].UserId), command.PlaceHolderWorkspace)
-		if serverError != nil {
-			err = serverError
-		}
-		return ok
-	})
-	if err != nil {
-		return nil, err
-	}
+	filteredTensorboards, err := a.filterTensorboards(ctx, *curUser, resp.Tensorboards)
```

Comment:
It seems like there is a missing piece on looking at what the user requested  `req.workspaceId` (something we might want to change into a list of workspace ids at some point)

or is this what you intended to do at a later pr? I think this affects how `AccessibleScopes` is defined so IMO it makes sense to define Accessible scopes at the same time that you start looking at this parameter. aka if you prefer to exclude `req.workspaceId` from this pr take out defining the `AccessibleScope` as well and add them together in a different pr